enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,490 @@
+
+In the first iteration, this can be implemented by defining a ServiceMonitor against the `/federate` endpoint of the Platform Monitoring Prometheus. Improvement on this side will be revisited in the next enhancement (e.g ability to scrape those data directly from sources).
+
+### High Availability
+
+All components except Prometheus are stateless and can run with more than one replica. In order to make Prometheus highly-available, a similar approach to the one in CMO will be used. The monitoring operator will provide a Thanos Querier for each `MonitoringStack` and a sidecar next to each Prometheus.
+A Thanos Ruler is not needed in this architecture since all recording rules and alerts will be evaluated locally in each Prometheus instance.
+
+### Alert Routing
+
+For simplicity reasons, routing alerts that come from managed service monitoring stacks will be handled by a single Alertmanager.
+We can take advantage of the recently added <code>[AlertmanagerConfig](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#alertmanagerconfig)</code> and allow MTO/MTSRE to create their own routing configuration in a centrally deployed Alertmanager instance.
+Finally, the MTSRE team could centrally configure receivers in one place for all managed services and service owners would simply use those receivers in their routing rules. It is worth noting that the last feature depends on [https://github.com/prometheus-operator/prometheus-operator/issues/4033](https://github.com/prometheus-operator/prometheus-operator/issues/4033).
+
+In addition to the built-in Alertmanager from MSO, users will have the option to route alerts to external Alertmanagers, such as the PM Alertmanager.
+
+#### Notable Alerting Routing Alternative
+
+During the Monitoring Enablement Working Group meetings, we received feedback from both Platform SRE and Managed Tenant SRE that Alertmanager, in general, has a low footprint, and the Prometheus Operator does a good job of upgrading it and keeping it running.
+
+Moreover, Platform SRE expressed concerns that a shared Alertmanager configuration could lead to conflicts or misconfiguration that can impact alerting for every managed tenant running in the cluster.
+
+Based on the information, we do not see strong reasons not to have one Alertmanager per managed service. Either approach should work as an initial solution.
+
+### Creating Dashboards
+
+The OCP Console does not (yet) support querying different Prometheus instances, making it hard to use it for creating dashboards.
+
+For this reason, the monitoring operator will optionally deploy a Grafana operator that can manage Grafana instances in the namespaces where a MonitoringStack CR is present. Managed service SREs and owners will then be able to use the CRDs provided by the Grafana operator to create their own dashboards.
```

Comment:
One Grafana instance per Addon? Again, need to think about the UX a bit here. Backplane already makes it hard enough to get into a single grafana.

Having to hop across Grafana instances during an incident could lead to some human errors and frustration.

Would it be possible to have a single Grafana and the Addons simply add their own datasource to this Grafana? 

This would allow for some use cases like local cluster-state dashboards that span several Addons.

Reply:
I thought we were removing grafana from OCP console ... now we are adding it back (albeit like a side car?) who owns this grafana operator for RHT?  Who supports it?

Reply:
We don't add it back to Core. It's similar support as MonitoringStack. We will support it together with other existing maintainers.

Reply:
Open question: 
* Do we then should descope Grafana (and similarly AM) from MonitoringStack CR and ask users to create AM CR (from PO) and Grafana instances (from Grafana Opetator) directly? It feels more natural and composable TBH.

Monitoring Stack can still create data sources CRs for Grafana integration.

Reply:
@aditya-konarde I agree that one Grafana per addon might not be ideal for MT-SRE. However, we would like to look at a `MonitoringStack` resource as independent of the Managed Services use case and even decoupled from OpenShift. In the future, we would like to be able to reuse this abstraction for different use cases.

If we allow end-users to deploy `MonitoringStack`s, their dashboards will automatically end up in this central Grafana instance. As an alternative, can MT-SRE simply deploy a new Grafana instance through the Grafana Operator and select datasources from all Addon namespaces. This would keep the `MonitoringStack` resource composable and reusable instead of coupling it to a single use case.

Reply:
I like this, but the problem I see is that MonitoringStack now is just a small wrapper/abstraction over Prometheus 🤔

Reply:
Which technically could be just a next PO CRD